src/operator/nn/mkldnn/mkldnn_convolution.cc

```diff
@@ -88,13 +88,25 @@ mkldnn::convolution_forward::primitive_desc GetConvFwdImpl(
   if (param.conv_param.dilate.ndim() == 0 && bias == nullptr) {
     mkldnn::convolution_forward::desc desc(prop, mkldnn::algorithm::convolution_direct,
         data_md, weight_md, out_md, strides, padding, padding, mkldnn::padding_kind::zero);
-    return mkldnn::convolution_forward::primitive_desc(desc, attr, engine);
+    auto conv_pd =  mkldnn::convolution_forward::primitive_desc(desc, attr, engine);
+    while (conv_pd.dst_primitive_desc().get_size() != GetArraySize(output) ||
+           conv_pd.src_primitive_desc().get_size() != GetArraySize(data) ||
+           conv_pd.weights_primitive_desc().get_size() != GetArraySize(weights)) {
+      CHECK(conv_pd.next_impl()) << "No implementation";
+    }
+    return conv_pd;
```

Comment:
please comment on what the change does.

Reply:
Also curious about this.  Is this an adaptation to an API change in MKLDNN?  Doe sit return a list of results now rather than a single value?

Reply:
Add comments. Yes, MKL-DNN introduced new API to select other implementations. `conv_pd` should have the same data type as before, not a list. `next_impl` is just a member function of it and can change some status of this object.